src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -1161,6 +1166,81 @@ void InterpreterSystemQuery::waitLoadingParts()
     }
 }
 
+
+void InterpreterSystemQuery::loadPrimaryKeys()
+{
+    if (!table_id.empty())
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY, table_id.database_name, table_id.table_name);
+        StoragePtr table = DatabaseCatalog::instance().getTable(table_id, getContext());
+
+        if (auto * merge_tree = dynamic_cast<MergeTreeData *>(table.get()))
+        {
+            LOG_TRACE(log, "Loading primary keys for table {}", table_id.getFullTableName());
+            try
+            {
+                merge_tree->loadPrimaryKeys();
+            }
+            catch (const Exception & ex)
+            {
+                LOG_ERROR(log, "Failed to load primary keys for table {}: {}", table_id.getFullTableName(), ex.message());
+            }
+        }
+        else
+        {
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS, "Command LOAD PRIMARY KEY is supported only for MergeTree tables, but got: {}", table->getName());
+        }
+    }
+    else
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY);
+
+        // Define a thread pool for parallel processing
+        auto & thread_pool = DB::getActivePartsLoadingThreadPool().get();
+
+        // Cap the number of concurrent tasks based on thread pool availability
+        size_t max_tasks = std::min(thread_pool.maxConcurrency(), DatabaseCatalog::instance().getDatabases().size());
+
+        // Process databases and tables in parallel
+        size_t tasks_scheduled = 0;
+        for (auto & database : DatabaseCatalog::instance().getDatabases())
+        {
+            for (auto it = database.second->getTablesIterator(getContext()); it->isValid(); it->next())
+            {
+                if (auto * merge_tree = dynamic_cast<MergeTreeData *>(it->table().get()))
+                {
+                    // Schedule task for thread pool
+                    if (tasks_scheduled < max_tasks)
+                    {
+                        thread_pool.scheduleOrThrowOnError([merge_tree, log=log] {
```

Comment:
The `log=log` raises a shadowing error:

```
ClickHouse/src/Interpreters/InterpreterSystemQuery.cpp:1216:73: error: declaration shadows a field of 'DB::InterpreterSystemQuery' [-Werror,-Wshadow-uncaptured-local]
 1216 |                         thread_pool.scheduleOrThrowOnError([merge_tree, log=log] {
```

You can either use a different name for the lambda variable, or capture `this`